File: client/generate-icons.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const iconSizes = [
  { size: 192, name: 'icon-192.png' },
  { size: 512, name: 'icon-512.png' },
  { size: 180, name: 'apple-touch-icon.png' },
  { size: 32, name: 'favicon-32x32.png' },
  { size: 16, name: 'favicon-16x16.png' }
];

const svgContent = fs.readFileSync(path.join(__dirname, 'public', 'app-icon.svg'), 'utf8');

iconSizes.forEach(({ size, name }) => {
  const modifiedSvg = svgContent
    .replace(/width="512"/, `width="${size}"`)
    .replace(/height="512"/, `height="${size}"`);

  const svgName = name.replace('.png', '.svg');
  fs.writeFileSync(path.join(__dirname, 'public', svgName), modifiedSvg);
});

console.log('Icons generated successfully!');

File: client/generate_icons.py
#!/usr/bin/env python3
"""
Generate PNG icons from SVG favicon for HyPockeTuner
"""

import subprocess
import os

# Ensure we're in the right directory
os.chdir('/home/donghee/HyPockeTuner_new/client/public')

# Icon sizes needed
sizes = [
    (16, 'favicon-16x16.png'),
    (32, 'favicon-32x32.png'),
    (48, 'icon-48.png'),
    (72, 'icon-72.png'),
    (144, 'icon-144.png'),
    (192, 'icon-192.png'),
    (192, 'logo192.png'),
    (256, 'icon-256.png'),
    (512, 'icon-512.png'),
    (512, 'logo512.png'),
    (180, 'apple-touch-icon.png'),
]

svg_file = 'favicon.svg'

for size, filename in sizes:
    try:
        # Use rsvg-convert if available, otherwise try inkscape
        try:
            cmd = [
                'rsvg-convert',
                '-w', str(size),
                '-h', str(size),
                svg_file,
                '-o', filename
            ]
            subprocess.run(cmd, check=True)
            print(f"✓ Generated {filename} ({size}x{size})")
        except (subprocess.CalledProcessError, FileNotFoundError):
            # Fallback to inkscape
            cmd = [
                'inkscape',
                svg_file,
                '--export-type=png',
                f'--export-filename={filename}',
                f'--export-width={size}',
                f'--export-height={size}'
            ]
            subprocess.run(cmd, check=True, capture_output=True)
            print(f"✓ Generated {filename} ({size}x{size}) using Inkscape")
    except Exception as e:
        print(f"✗ Failed to generate {filename}: {e}")

print("\nIcon generation complete!")
print("Remember to rebuild the app with 'npm run build' to see the changes.")